File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/repository/PostgresRepository.java
package com.fixture.shop.repository;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fixture.shop.models.Order;
import com.fixture.shop.models.OrderStatus;
import com.fixture.shop.models.Product;

/**
 * Postgres-backed implementation of Repository.
 */
public class PostgresRepository implements Repository {

    private final String connectionString;
    private final Map<String, Order> orders = new HashMap<>();
    private final Map<String, Product> products = new HashMap<>();

    public PostgresRepository(String connectionString) {
        this.connectionString = connectionString;
    }

    @Override
    public void saveOrder(Order order) {
        orders.put(order.getOrderId(), order);
    }

    @Override
    public Order getOrder(String orderId) {
        return orders.get(orderId);
    }

    @Override
    public void updateOrderStatus(String orderId, OrderStatus status) {
        Order order = orders.get(orderId);
        if (order != null) {
            order.setStatus(status);
        }
    }

    @Override
    public Product getProduct(String sku) {
        return products.get(sku);
    }

    @Override
    public List<Order> listOrdersByUser(String userId) {
        List<Order> result = new ArrayList<>();
        for (Order order : orders.values()) {
            if (order.getUserId().equals(userId)) {
                result.add(order);
            }
        }
        return result;
    }

    public String getConnectionString() {
        return connectionString;
    }
}

File: skill_sdk/tests/fixtures/cpp-project/src/shop/PostgresRepository.cpp
#include "PostgresRepository.h"

PostgresRepository::PostgresRepository(const std::string& connStr) : connStr_(connStr) {}

bool PostgresRepository::SaveOrder(const Order& order) {
    orders_[order.OrderID] = order;
    return true;
}

Order PostgresRepository::GetOrder(const std::string& orderID) {
    auto it = orders_.find(orderID);
    return it != orders_.end() ? it->second : Order{};
}

bool PostgresRepository::UpdateOrderStatus(const std::string& orderID, OrderStatus status) {
    auto it = orders_.find(orderID);
    if (it == orders_.end()) return false;
    it->second.Status = status;
    return true;
}

Product PostgresRepository::GetProduct(const std::string& sku) {
    auto it = products_.find(sku);
    return it != products_.end() ? it->second : Product{};
}

std::vector<Order> PostgresRepository::ListOrdersByUser(const std::string& userID) {
    std::vector<Order> result;
    for (const auto& [_, order] : orders_) {
        if (order.UserID == userID)
            result.push_back(order);
    }
    return result;
}
